NeoML/test/src/DnnHeadTest.cpp

```diff
@@ -0,0 +1,263 @@
+/* Copyright © 2024 ABBYY
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#include <common.h>
+#pragma hdrstop
+
+#include <NeoML/Dnn/DnnHead.h>
+#include <NeoML/Dnn/Layers/DnnHeadAdapterLayer.h>
+#include <TestFixture.h>
+
+using namespace NeoML;
+using namespace NeoMLTest;
+
+TEST( CDnnHeadTest, CDnnHeadInferenceAndLearnTest )
+{
+    //        +----------------+
+    //        |                |
+    //        |                v
+    //    [source1]        |-----------------------------------------|
+    //                     | [Fc1]-->[Gelu]-->[Fc2]-->[Relu]-->[Fc3] | -> [Concat] ->[Loss]
+    //    [source2]        |-----------------------------------------|
+    //        |                ^
+    //        |                |
+    //        +----------------+
+
+    IMathEngine& mathEngine = MathEngine();
+    CRandom random( 0 );
+    CDnn dnn( random, mathEngine );
+
+    CPtr<CHeadStruct> head = CDnnHead<
+        CFullyConnectedLayer,
+        CGELULayer,
+        CFullyConnectedLayer,
+        CReLULayer,
+        CFullyConnectedLayer
+    >(
+        random, mathEngine,
+        FullyConnected( 500 ),
+        Gelu(),
+        FullyConnected( 100 ),
+        Relu(),
+        FullyConnected( 1 )
+    );
+
```

Comment:
Can you unite the common part of these tests?
It better to see for the performance-tests that code is the same except a small part, but the results are different.

Please, add

* several simple check-consistance tests of CDnnHeadAdapterLayer: the user test, a head inside a recurentLayer, some others, results of CDnnHeadAdapterLayer should be compared with network without it, and should be the same
* 2 performance tests with/without CDnnHeadAdapterLayer

performance tests should be disabled by default, like in loraTest
DISABLED_\<TestName\>